lib/addons/upbound-universal-crossplane/index.ts

```diff
@@ -1,36 +1,96 @@
+import {ClusterInfo, Values} from "../../spi";
+import { merge } from "ts-deepmerge";
+import {createNamespace, supportsALL} from '../../utils';
+import { Policy, PolicyDocument} from 'aws-cdk-lib/aws-iam';
+import * as cdk from 'aws-cdk-lib';
+import {HelmAddOn, HelmAddOnProps, HelmAddOnUserProps} from "../helm-addon";
 
 /**
- * Interface for Upbound Universal Crossplane EKS add-on options
+ * User provided options for the Helm Chart
  */
-interface UpboundUniversalCrossplaneAddOnProps {
+export interface UpboundCrossplaneAddOnProps extends HelmAddOnUserProps {
     /**
-     * Version of the driver to deploy
+     * To Create Namespace using CDK
      */
-    version?: string;
+    createNamespace?: boolean;
+    /*
+     * EKS MasterRole. `mastersRole` of blueprints should be passed to this parameter.
+     */
+    eksMasterRole?: any;
 }
 
-/**
- * Default values for the add-on
- */
 const defaultProps = {
-    addOnName: "upbound_universal-crossplane",
-    version: "v1.9.1-eksbuild.0"
+    name: 'uxp',
+    release: 'blueprints-addon-uxp',
+    namespace: 'upbound-system',
+    chart: 'universal-crossplane',
+    version: '1.14.5-up.1',
+    repository: 'https://charts.upbound.io/stable',
+    values: {},
+    eksMasterRole: "Admin"
 };
 
-/**
- * Implementation of Upbound Crossplane EKS add-on
- */
-@supportsX86
-export class UpboundUniversalCrossplaneAddOn extends CoreAddOn {
-
-    constructor(readonly options?: UpboundUniversalCrossplaneAddOnProps) {
-        super({
-            addOnName: defaultProps.addOnName,
-            version: options?.version ?? defaultProps.version,
-            saName: ""
-        });
+@supportsALL
+export class UpboundCrossplaneAddOn extends HelmAddOn {
+
+    readonly options: UpboundCrossplaneAddOnProps;
+
+    constructor( props?: UpboundCrossplaneAddOnProps) {
+        super({...defaultProps, ...props});
+
+        this.options = this.props as UpboundCrossplaneAddOnProps;
     }
-}
 
+    deploy(clusterInfo: ClusterInfo): void | Promise<Construct> {
+        const cluster = clusterInfo.cluster;
+
+        // Create the `upbound-system` namespace.
+        const ns = createNamespace(this.options.namespace!, cluster, true);
+
+        // Create the CrossPlane AWS Provider IRSA.
+        const serviceAccountName = "provider-aws";
+        const sa = cluster.addServiceAccount(serviceAccountName, {
+            name: serviceAccountName,
+            namespace: this.options.namespace!,
+
+        });
+
+        sa.node.addDependency(ns);
+        sa.role.attachInlinePolicy(new Policy(cluster.stack, 'eks-connect-policy',  {
+            document: PolicyDocument.fromJson({
+                "Version": "2012-10-17",
+                "Statement": [
+                    {
+                        "Effect": "Allow",
+                        "Action": ["sts:AssumeRole"],
+                        "Resource": `arn:aws:iam::${cluster.stack.account}:role/${this.options.eksMasterRole}`

```

Comment:
I suppose the policy should be modified to assume masters role associated with clusters that will under this cluster control. I think the design looks good - customers can create a single role and make it the masters role for each cluster they want to manage with crossplane. That is what we used in MP/Conformitron. 
What is missing is this description in the doc.

Reply:
Agreed, description of the doc is updated and also updated the overall doc,.